File: src/common/py/core/messaging/composers/event_composer.py
from .message_composer import MessageComposer
from ..meta import MessageMetaInformation


class EventComposer(MessageComposer):
    """
    Composer for ``Event`` messages.
    """

    def _create_meta_information(
        self, suppress_logging: bool
    ) -> MessageMetaInformation:
        from ..meta import EventMetaInformation

        return EventMetaInformation(
            entrypoint=MessageMetaInformation.Entrypoint.LOCAL,
            suppress_logging=suppress_logging,
        )

File: src/common/web/core/messaging/composers/EventComposer.ts
import { Event } from "../Event";
import { EventMetaInformation } from "../meta/EventMetaInformation";
import { MessageEntrypoint, MessageMetaInformation } from "../meta/MessageMetaInformation";
import { MessageComposer } from "./MessageComposer";

/**
 * Composer for ``Event`` messages.
 */
export class EventComposer<MsgType extends Event> extends MessageComposer<MsgType> {
    protected createMetaInformation(): MessageMetaInformation {
        return new EventMetaInformation(MessageEntrypoint.Local);
    }
}
